File: app/src/main/java/cz/inventi/inventiskeleton/presentation/post/add/PostAddComponent.java
package cz.inventi.inventiskeleton.presentation.post.add;

import cz.inventi.inventiskeleton.di.ScreenModule;
import cz.inventi.inventiskeleton.di.ScreenScope;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

/**
 * Created by tomas.valenta on 5/11/2017.
 */

@ScreenScope @Subcomponent(modules = { ScreenModule.class })
public interface PostAddComponent extends AndroidInjector<PostAddController> {

    @Subcomponent.Builder abstract class Builder extends AndroidInjector.Builder<PostAddController> {
        public abstract Builder screenModule(ScreenModule screenModule);

        @Override public void seedInstance(PostAddController postAddController) {
            screenModule(new ScreenModule(postAddController.getClass().getSimpleName()));
        }
    }
}

File: app/src/main/java/cz/inventi/inventiskeleton/presentation/post/add/PostAddComponent.kt
package cz.inventi.inventiskeleton.presentation.post.add

import cz.inventi.inventiskeleton.di.ScreenModule
import cz.inventi.inventiskeleton.di.ScreenScope
import dagger.Subcomponent
import dagger.android.AndroidInjector

/**
 * Created by tomas.valenta on 5/11/2017.
 */

@ScreenScope @Subcomponent(modules = arrayOf(ScreenModule::class))
interface PostAddComponent : AndroidInjector<PostAddController> {

    @Subcomponent.Builder abstract class Builder : AndroidInjector.Builder<PostAddController>() {
        abstract fun screenModule(screenModule: ScreenModule): Builder

        override fun seedInstance(postAddController: PostAddController) {
            screenModule(ScreenModule(postAddController.javaClass.simpleName))
        }
    }
}
